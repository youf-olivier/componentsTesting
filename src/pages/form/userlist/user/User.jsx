import React from 'react';
import PropTypes, { shape } from 'prop-types';

import './User.scss';

export const userType = {
  html_url: PropTypes.string.isRequired,
  avatar_url: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
};

const propTypes = {
  user: shape(userType).isRequired,
};

const UserCard = ({ user }) => (
  <div className="user-card" role="listitem" aria-label="carte utilisateur">
    <a href={user.html_url} className="user-card__link">
      <img src={user.avatar_url} className="user-card__avatar" alt={`${user.login}`} />
      <span className="user-card__login">{user.login}</span>
    </a>
  </div>
);

UserCard.propTypes = propTypes;

export default UserCard;
