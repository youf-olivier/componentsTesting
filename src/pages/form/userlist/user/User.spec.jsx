import React from 'react';
import { render } from '@testing-library/react';
import User from './User';

describe('User card test suite', () => {
  const user = {
    login: '123123',
    avatar_url: 'http://avatar.jpg',
    html_url: 'http://comptegithub.com',
  };

  it('renders correctly', () => {
    const { asFragment } = render(<User user={user} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
