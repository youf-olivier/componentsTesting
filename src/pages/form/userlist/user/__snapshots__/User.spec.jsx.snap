// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User card test suite renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="carte utilisateur"
    class="user-card"
    role="listitem"
  >
    <a
      class="user-card__link"
      href="http://comptegithub.com"
    >
      <img
        alt="123123"
        class="user-card__avatar"
        src="http://avatar.jpg"
      />
      <span
        class="user-card__login"
      >
        123123
      </span>
    </a>
  </div>
</DocumentFragment>
`;
